hibernate-cache/src/test/java/org/infinispan/test/hibernate/cache/tm/JBossStandaloneJtaExampleTest.java

```diff
@@ -108,12 +108,10 @@ public void testPersistAndLoadUnderJta() throws Exception {
 			ut.begin();
 			try {
 				Session session = sessionFactory.openSession();
-				assertTrue(session.getTransaction().isActive());
+            session.getTransaction().begin();
 				item = new Item("anItem", "An item owned by someone");
 				session.persist(item);
-				// IMO the flush should not be necessary, but session.close() does not flush
-				// and the item is not persisted.
-				session.flush();
+            session.getTransaction().commit();
```

Comment:
The idea is that you close the session before JTA transaction is committed. Besides, this wouldn't use the correct transaction handling pattern.